File: No_2598/Smallest Missing Non-negative Integer After Operations.cpp
class Solution {
public:
    int findSmallestInteger(vector<int>& nums, int value) {
        unordered_map<int, int> count;
        
        for (int num : nums) {
            count[(num % value + value) % value]++;
        }
        
        int modValue = 0;
        for (int i = 0; i < value; i++) {
            if (count[i] < count[modValue])
                modValue = i;
        }
        
        return modValue + (value * count[modValue]);
    }
};

File: No_2598/Smallest Missing Non-negative Integer After Operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        count = defaultdict(int)
        
        for i in range(value):
            count[i] = 0
            
        for num in nums:
            count[num % value] += 1
            
        min_item = min(count.items(), key=lambda item: (item[1], item[0]))
        
        return min_item[0] + (value * min_item[1])